assets/src/edit-story/components/dropTargets/provider.js

```diff
@@ -172,6 +172,23 @@ function DropTargetsProvider({ children }) {
         });
 
       setActiveDropTargetId(null);
+
+      // Auto-play video on drop
+      if (resource.type === 'video') {
+        // eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+        setTimeout(() => {
+          const videoEl = document.getElementById(
+            `video-${activeDropTargetId}`
+          );
+          if (videoEl) {
+            videoEl.load();
+            const playPromise = videoEl.play();
+            if (playPromise) {
+              playPromise.catch(() => {});
+            }
+          }
+        });
+      }
```

Comment:
Could we perhaps move this to the video element definition? As some kind of `onDropHandler` returned by `getDefinitionForType`?  This feels a bit dirty and malplaced here.

Reply:
Done in https://github.com/google/web-stories-wp/pull/2286/commits/121d8a6781a396c04a8349dd6007a260f1c59cfc